### PR TITLE
Mob/moderation list view

### PR DESCRIPTION
**Pull request by fabmiz**
- Ticket: []
- Feature flag: n/a

### CODE REVIEW

File: lib/registries/addon/components/registration-list/card/component.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import { action } from '@ember/object';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { task } from 'ember-concurrency-decorators';
+
+import RegistrationModel from 'ember-osf-web/models/registration';
+import ReviewActionModel from 'ember-osf-web/models/review-action';
+
+const iconMap: Record<string, string> = {
```
Comment (fabmiz) [inline lib/registries/addon/components/registration-list/card/component.ts:9]: Use `RegistrationReviewStates`
```suggestion
const iconMap: Record< RegistrationReviewStates, string> = {
```

File: lib/registries/addon/components/registration-list/card/component.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    pending: 'hourglass',
+    withdrawn: 'ban',
+    accepted: 'check',
+    rejected: 'times',
+    pending_withdraw: 'clock-o',
+};
+
+interface Args {
+    registration: RegistrationModel;
+    filterState: string;
```
Comment (fabmiz) [inline lib/registries/addon/components/registration-list/card/component.ts:19]:
```suggestion
    filterState: RegistrationReviewStates;
```

File: lib/registries/addon/components/registration-list/card/template.hbs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        data-test-registration-list-card-icon='{{@filterState}}'
+        @icon={{this.icon}}
+        @size={{2}}
+        local-class='filterIcon {{@filterState}}'
+    />
+    {{#if this.fetchActions.isRunning}}
+        <LoadingIndicator />
+    {{else}}
+        <div local-class='cardInfo'>
+            {{!-- TODO: Update this to be a link to the moderation overview page --}}
```
Comment (fabmiz) [inline lib/registries/addon/components/registration-list/card/template.hbs:16]: Address this ToDo item

File: mirage/factories/review-action.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+declare module 'ember-cli-mirage/types/registries/model' {
+    export default interface MirageModelRegistry {
+        'review-action': ReviewActionModel;
+    } // eslint-disable-line semi
+}
+
+declare module 'ember-cli-mirage/types/registries/schema' {
+    export default interface MirageSchemaRegistry {
+        reviewAction: ReviewActionModel;
```
Comment (fabmiz) [inline mirage/factories/review-action.ts]: Pluralize
```suggestion
        reviewActions: ReviewActionModel;
```

File: translations/en-us.yml [hunk 5]
```diff
@@ -1259,6 +1284,14 @@ registries:
...
         selectAll: 'Select All'
         clearAll: 'Clear All'
+    registrationList:
+        pending: 'No registrations pending approval have been found'
+        accepted: 'No accepted registrations have been found'
+        rejected: 'No rejected registrations have been found'
+        withdrawn: 'No withdrawn registrations have been found'
+        pending_withdraw: 'No registrations found pending withdrawal'
+        moderatorAction: '{action} {date} by {moderator}'
+        noActionsFound: 'Failed to load registration review history'
```
Comment (fabmiz) [inline translations/en-us.yml:1294]: some consistency would be nice?
```suggestion
        moderator_action: '{action} {date} by {moderator}'
        no_actions_found: 'Failed to load registration review history'
```

File: mirage/serializers/user.ts [hunk 6]
```diff
@@ -5,13 +5,13 @@ import User from 'ember-osf-web/models/user';
...
 
-import ApplicationSerializer from './application';
+import ApplicationSerializer, { SerializedRelationships } from './application';
 
 const { OSF: { apiUrl } } = config;
 
 export default class UserSerializer extends ApplicationSerializer<User> {
     buildRelationships(model: ModelInstance<User>) {
-        return {
+        const retVal: SerializedRelationships<User> = {
```
Comment (fabmiz) [inline mirage/serializers/user.ts:14]: minor: `retVal` is not a good variable name
```suggestion
        const serializedRelationships: SerializedRelationships<User> = {
```

File: mirage/serializers/registration.ts [hunk 7]
```diff
@@ -123,12 +123,36 @@ export default class RegistrationSerializer extends ApplicationSerializer<Mirage
...
+                },
+            };
+        }
+
         if (model.attrs.providerId) {
             const { providerId } = model.attrs;
             relationships.provider = {
                 data: {
-                    id: providerId,
+                    id: model.attrs,
```
Comment (fabmiz) [inline mirage/serializers/registration.ts:155]: why was this changed?
```suggestion
                    id: providerId,
```

**Review by fabmiz — COMMENTED** [review]
overall great works 💯

File: lib/registries/addon/components/registration-list/card/template.hbs (hunk 3 above)
Comment (fabmiz) [reply lib/registries/addon/components/registration-list/card/template.hbs:16]: Will get done once moderation overview page is in.

File: lib/registries/addon/overview/-components/overview-header/component.ts [hunk 8]
```diff
@@ -1,9 +1,58 @@
...
+        try {
+            this.currentModerator = yield this.store.findRecord('moderator', this.currentUser.currentUserId!,
+                {
+                    adapterOptions: {
+                        providerId: this.registration.provider.get('id'),
+                    },
+                });
+        } catch (e) {
+            captureException(e);
+            this.toast.error(getApiErrorMessage(e));
```
Comment (futa-ikeda) [inline lib/registries/addon/overview/-components/overview-header/component.ts:49]: Change this toast message to tell the user they need moderator permissions

File: lib/registries/addon/overview/-components/overview-header/template.hbs [hunk 9]
```diff
@@ -3,9 +3,9 @@
         <div local-class='Header'>
             <h1 data-test-registration-title>{{@registration.title}}</h1>
         </div>
-        {{#if (not (or @registration.archiving @registration.withdrawn))}}
+        {{#if (not (or @registration.archiving (eq @registration.reviewsState 'withdrawn')))}}
```
Comment (futa-ikeda) [inline lib/registries/addon/overview/-components/overview-header/template.hbs:6]: let's change this to a computed property

File: lib/registries/addon/overview/controller.ts [hunk 10]
```diff
@@ -18,8 +19,11 @@ const { OSF: { url: baseURL } } = config;
 export default class Overview extends Controller {
     model!: GuidRouteModel<Registration>;
 
+    queryParams = ['viewMode'];
```
Comment (futa-ikeda) [inline lib/registries/addon/overview/controller.ts:22]: call this `mode`?

File: lib/registries/addon/overview/template.hbs [hunk 11]
```diff
@@ -9,13 +9,13 @@
     {{else}}
         {{assert 'Registries::Overview - registration should not be null or undefined' this.registration}}
         {{title this.registration.title prepend=false}}
-        {{#if (or this.registration.withdrawn this.registration.archiving)}}
+        {{#if (or (eq this.registration.reviewsState 'withdrawn') this.registration.archiving)}}
```
Comment (futa-ikeda) [inline lib/registries/addon/overview/template.hbs:12]: make this computed as well?

File: translations/en-us.yml [hunk 12]
```diff
@@ -977,6 +980,34 @@ registries:
...
+        page_title: 'Registries'
+    moderation:
+        header: Moderation
+        states:
+            accepted: Public
+            embargo: Embargo
+            rejected: Rejected
+            pending: Pending
+            withdrawn: Withdrawn
+            pendingWithdraw: 'Pending Withdraw'
```
Comment (futa-ikeda) [inline translations/en-us.yml:993]:
```suggestion
            pendingWithdraw: 'Pending Withdrawal'
```

**Review by futa-ikeda — CHANGES_REQUESTED** [review]

**Review by futa-ikeda — DISMISSED** [review]

**Review by futa-ikeda — APPROVED** [review]

**Review by adlius — APPROVED** [review]

**Pull request merged**